### PR TITLE
Code improvement to configuration file load process

### PR DESCRIPTION
**Pull request by khalidmammadov**
This improves configuration file load process by following:
- Removes code duplication
- Introduce single function that can load any config and provides default one when no config files specified
- Changes `conf` local variable to `_conf` to avoid confusion between global variable and local one

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
I am always afraid to touch this part of code. You never know what's gonna blow when you do.

Let's wait for the tests to complete. Did you actually try to run some basic airflow commands @khalidmammadov  - from my experience changes in conf often lead to airflow failing to start :D

**Comment by khalidmammadov** [comment]
I have run whole test suite before and after change and no regression was detected. Also, run airflow scheduler and webserver and some dag runs and commands, again no issues.
To be clear the change only about code improvement and no logic was changed, I did make sure it does exactly the same things but with no duplicate code and hopefully bit more clear to understand/follow along.

**Comment by potiuk** [comment]
> To be clear the change only about code improvement and no logic was changed, I did make sure it does exactly the same things but with no duplicate code and hopefully bit more clear to understand/follow along.

Cool! Anyone else can review and confirm (we have the rule those kind of change - to the core - need two committers approvals @khalidmammadov )

**Review by potiuk — APPROVED** [review]

**Pull request closed without merging**
